aiplatform/api/AIPlatform.Samples/PredictChatPromptSample.cs

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START aiplatform_sdk_chat]
+
+using Google.Cloud.AIPlatform.V1;
+using wkt = Google.Protobuf.WellKnownTypes;
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+public class PredictChatPromptSample
+{
+    public PredictResponse PredictChatPrompt(
+        string projectId = "your-project-id",
+        string locationId = "us-central1",
+        string publisher = "google",
+        string model = "chat-bison@001"
+    )
+    {
+        // Initialize client that will be used to send requests.
+        // This client only needs to be created
+        // once, and can be reused for multiple requests.
+        var client = new PredictionServiceClientBuilder
+        {
+            Endpoint = $"{locationId}-aiplatform.googleapis.com"
+        }.Build();
+
+        // Configure the parent resource
+        var endpoint = EndpointName.FromProjectLocationPublisherModel(projectId, locationId, publisher, model).ToString();
+
+        // Initialize request argument(s)
+        var prompt = "How many planets are there in the solar system?";
+
+        // You can construct Protobuf from JSON
+        var instanceJson = JsonConvert.SerializeObject(new
+        {
+            context = "My name is Miles. You are an astronomer, knowledgeable about the solar system.",
+            examples = new[]
+            {
+                new
+                {
+                    input = new { content = "How many moons does Mars have?" },
+                    output = new { content = "The planet Mars has two moons, Phobos and Deimos." }
+                }
+            },
+            messages = new[]
+            {
+                new
+                {
+                    author = "user",
+                    content = prompt
+                }
+            }
+        });
```

Comment:
I like the idea of showing both methods, but it seems a little overwhelming to have so much in this one sample. Maybe stick to one or the other?

Reply:
Hi @holtskinner, I agree it's a little overwhelming but Protobuf is really impossible to construct on your own and JSON-to-Protobuf is brittle. I wanted to have one sample where people see both and decide which one to use on their own. In other samples, I'll probably stick with Protobuf only but it'd be useful to have one sample to show both ways, so people can make the link between the two. What do you think?

Reply:
Good to get the context on this. Sharing my thoughts - Given it is a sample, which should  be clear, easy to understand and simple I would prefer to use the Protobuf directly in sample.  If we want to show the other two approaches we can possibly add it in the documentation page or add the corresponding approaches in readme.md and paste its link in the sample. That way we can keep this sample simple and clear and also provide alternatives. Ones the library is out with correct protos, we will need to revisit this, so we can also add a TODO.   This applies to both instances and parameters.